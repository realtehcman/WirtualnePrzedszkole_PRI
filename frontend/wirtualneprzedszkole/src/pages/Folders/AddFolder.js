import React, { useEffect, useState} from 'react'
import {useNavigate} from "react-router-dom";
import FolderService from './FolderService'
import '../CreateUser/CreateUser.scss'
import './AddFolder.scss'
import FileService from "../gallery/FileService"

const AddFolder = (props) => {
    const [newFolder, setNewFolder] = useState(
        {
            name: "",
            path: "",
            className: "",
            fileDataList: [{}],
        }
    )
    
    let {folder} = props


    const addFolder = async(event) => {
        event.preventDefault();
        const formData = new FormData(event.currentTarget);
        
        const files = event.currentTarget;
        for (let i = 1; i < files.length; i++) {
            formData.append('file', files[i]);
        }
        newFolder.path = folder.path + "/" + newFolder.name
        newFolder.className = folder.className
        let temp = formData.getAll("file").length
        console.log(temp)
        FolderService.addNewFolder(newFolder).then(response => {
            console.log(response.data)
            if (temp > 3)
                FileService.addFiles(response.data.id, formData)
        })
    }

    return (
        <div className='formContainer'>
            <div className='row'>
                <div className='card col-md-6 offset-md-3 offset-md-3'>
                    <div className='form-body'>
                        <form onSubmit={addFolder} encType='multipart/form-data'>
                            <div className='form-group'>
                                <input placeholder='Nazwa Folderu' name="Nazwa Folderu" className='"form-control' 
                                onChange={e => setNewFolder({name: e.target.value})}/>
                            </div>
                            <div className="uploadDiv2">
                                    <div className="input25">   <input type="file" className="form-control" id="customFile" name='file' multiple/></div>
                            </div>
                            <div className="form-but">
                                    <button className="button">Zapisz</button>
                            </div>
                        </form>
                    </div>
                </div>

            </div>
        </div>
    )
}

export default AddFolder