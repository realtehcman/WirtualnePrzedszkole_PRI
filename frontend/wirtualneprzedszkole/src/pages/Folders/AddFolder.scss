.input25{
    max-width: 100%;
  }

  .uploadDiv2 {
    margin-left: 5%;
    margin-right: 5%;
  }