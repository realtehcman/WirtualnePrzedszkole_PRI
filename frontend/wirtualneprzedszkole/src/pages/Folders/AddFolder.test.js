import React from 'react'
import {configure} from 'enzyme'
import AddFolder from './AddFolder'
import Adapter from "enzyme-adapter-react-16";
import {MemoryRouter} from "react-router-dom";
import {render} from "@testing-library/react";

configure({adapter: new Adapter()});

jest.mock('./FolderService')
jest.mock("../gallery/FileService")

describe('AddFolder component', () => {
    it('renders correctly', () => {
        const {getByTestId} = render(
            <MemoryRouter>
                <AddFolder/>
            </MemoryRouter>
        );
        expect(getByTestId('add-folder')).toBeTruthy();
    });
});
