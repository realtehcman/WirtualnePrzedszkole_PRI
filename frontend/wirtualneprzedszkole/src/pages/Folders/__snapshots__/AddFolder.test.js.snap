// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFolder should render correctly 1`] = `ShallowWrapper {}`;
